ethereum/eth/src/main/java/org/hyperledger/besu/ethereum/eth/transactions/TransactionsForSenderInfo.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.hyperledger.besu.ethereum.eth.transactions;
+
+import java.util.Iterator;
+import java.util.PriorityQueue;
+import java.util.Queue;
+import java.util.SortedMap;
+import java.util.TreeMap;
+import java.util.stream.LongStream;
+
+class TransactionsForSenderInfo {
+  private final SortedMap<Long, PendingTransactions.TransactionInfo> transactionsInfos;
+  private final Queue<Long> gaps = new PriorityQueue<>();
+
+  TransactionsForSenderInfo() {
+    transactionsInfos = new TreeMap<>();
+  }
+
+  void addTransactionToTrack(
+      final long nonce, final PendingTransactions.TransactionInfo transactionInfo) {
+    if (!transactionsInfos.isEmpty()) {
+      updateGapsOnNewTransaction(nonce);
+    }
+    transactionsInfos.put(nonce, transactionInfo);
+  }
+
+  private void updateGapsOnNewTransaction(final long nonce) {
+    final long highestNonce = transactionsInfos.lastKey();
+    if (nonce > (highestNonce + 1)) {
```

Comment:
I'm not sure this `if` is needed, I think `LongStream` might handle that in the case that the first is larger than the second

Reply:
nice catch. fixed.